org.jacoco.core/src/org/jacoco/core/internal/analysis/MethodAnalyzer.java

```diff
@@ -86,6 +99,38 @@ public IMethodCoverage getCoverage() {
 		return coverage;
 	}
 
+	/**
+	 * {@link MethodNode#accept(MethodVisitor)}
+	 */
+	@Override
+	public void accept(final MethodNode methodNode,
+			final MethodVisitor methodVisitor) {
+		this.ignored.clear();
+		for (final IFilter filter : FILTERS) {
+			filter.filter(methodNode, this);
+		}
+
+		for (final TryCatchBlockNode n : methodNode.tryCatchBlocks) {
+			n.accept(methodVisitor);
+		}
+		for (int i = 0; i < methodNode.instructions.size(); i++) {
```

Comment:
@marchof as link in Javadoc shows - this was inspired by implementation of `org.objectweb.asm.tree.MethodNode#accept(org.objectweb.asm.MethodVisitor)`, so strictly speaking there is no change in behavior in regards to creation of array. But you right that we can avoid this. And if so, then I'm wondering - how to prevent regression of this? Probably it can/should be avoided for instrumentation too? Are there other places that trigger or might trigger its creation?